dbt/adapters/spark/connections.py

```diff
@@ -278,7 +289,7 @@ class SparkConnectionManager(SQLConnectionManager):
 
     SPARK_CLUSTER_HTTP_PATH = "/sql/protocolv1/o/{organization}/{cluster}"
     SPARK_SQL_ENDPOINT_HTTP_PATH = "/sql/1.0/endpoints/{endpoint}"
-    SPARK_CONNECTION_URL = "{host}:{port}" + SPARK_CLUSTER_HTTP_PATH
+    SPARK_CONNECTION_URL = "https://{host}:{port}" + SPARK_CLUSTER_HTTP_PATH
```

Comment:
Hmm, wasn't that already the case before, with this code?

```python

                    # Prepend https:// if it is missing
                    host = creds.host
                    if not host.startswith("https://"):
                        host = "https://" + creds.host
```

I tried to "standardize" the host name in `SparkCredentials. __post_init__()`, so it wouldn't matter if you added the prefix and / or the trailing `/`, and then added this here, because you had the https:// prefix for the `http` connection method